subprojects/snapshots/src/main/java/org/gradle/internal/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -192,12 +196,7 @@ public void handleChange(FileWatcherRegistry.Type type, Path path) {
 
                 @Override
                 public void handleLostState() {
-                    try {
-                        fileEvents.put(new FileEvent(true));
-                    } catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                        throw new RuntimeException(e);
-                    }
+                    fileEvents.offer(new FileEvent(true));
```

Comment:
Here we need to make sure the event is handled, right? In either case we should handle the lost state the same way, so it should be extracted into a shared method, too.

Also, we should log something when we lose state. We should also name the reason for the lost state in the log message, given that now we can have two different reasons.